File: exercism.io/python/space-age/space_age.py
class SpaceAge(object):
    def __init__(self, seconds):
        self.earth_years = seconds / 60 / 60 / 24 / 365.25
        pass

    def on_mercury(self):
        return float("{0:.2f}".format(self.earth_years / 0.2408467, 2))

    def on_venus(self):
        return float("{0:.2f}".format(self.earth_years / 0.61519726, 2))
    
    def on_earth(self):
        return float("{0:.2f}".format(self.earth_years, 2))

    def on_mars(self):
        return float("{0:.2f}".format(self.earth_years / 1.8808158, 2))

    def on_jupiter(self):
        return float("{0:.2f}".format(self.earth_years / 11.862615, 2))

    def on_saturn(self):
        return float("{0:.2f}".format(self.earth_years / 29.44749, 2))

    def on_uranus(self):
        return float("{0:.2f}".format(self.earth_years / 84.016846, 2))

    def on_neptune(self):
        return float("{0:.2f}".format(self.earth_years / 164.79132, 2))

File: exercism.io/java/space-age/src/main/java/SpaceAge.java
class SpaceAge {

    private double seconds = 0;
    private int earthYear = 31557600;

    SpaceAge(double seconds) {
        this.seconds = seconds;
    }

    double onEarth() {
        return seconds / earthYear;
    }

    double onMercury() {
        return seconds / (earthYear * 0.2408467);
    }

    double onVenus() {
        return seconds / (earthYear * 0.61519726);
    }

    double onMars() {
        return seconds / (earthYear * 1.8808158);
    }

    double onJupiter() {
        return seconds / (earthYear * 11.862615);
    }

    double onSaturn() {
        return seconds / (earthYear * 29.447498);
    }

    double onUranus() {
        return seconds / (earthYear * 84.016846);
    }

    double onNeptune() {
        return seconds / (earthYear * 164.79132 );
    }

}
